llvm/lib/Bitcode/Writer/BitcodeWriter.cpp

```diff
@@ -4349,9 +4366,11 @@ static void writeFunctionHeapProfileRecords(
       Record.push_back(AI.Versions.size());
     for (auto &MIB : AI.MIBs) {
       Record.push_back((uint8_t)MIB.AllocType);
-      // Record the index into the radix tree array for this context.
-      assert(CallStackCount <= CallStackPos.size());
-      Record.push_back(CallStackPos[CallStackCount++]);
+      if (PerModule || CombinedIndexMemProfContext) {
```

Comment:
Can you explain a bit about the PerModule usage here and below?

Reply:
Added a comment here. The below usage was pre-existing so I wasn't sure another comment was needed.